Src/ReadCombustionTankPressure.c

```diff
@@ -0,0 +1,85 @@
+#include "stm32f4xx.h"
+#include "stm32f4xx_hal_conf.h"
+#include "cmsis_os.h"
+#include "math.h"
+
+#include "ReadCombustionTankPressure.h"
+
+#include "Data.h"
+
+#define ADC1_QUEUE_SIZE 5
+
+static int READ_COMBUSTION_TANK_PRESSURE_PERIOD = 300;
+
+static const int ADC1_POLL_TIMEOUT = 150;
+
+static uint16_t adc1ValuesQueue[ADC1_QUEUE_SIZE] = {0};    // Average ADC1_QUEUE_SIZE ADC readings that are each 12 bits in size
+// ( maximum resolution for this ADC )
+
+void readCombustionTankPressureTask(void const* arg)
+{
+    CombustionTankPressureData* data = (CombustionTankPressureData*) arg;
+    uint32_t prevWakeTime = osKernelSysTick();
+
+    double vo = 0;  // The voltage across the 133k resistor
+    double vi = 0;  // The pressure sensor output
+    double tankPressure = 0;
+
+    int adc1QueueIndex = 0;    // Counts up to ADC1_QUEUE_SIZE ADC readings before averaging
+
+    // Resistor values in kOhms
+    int R1 = 100;
+    int R2 = 133;
+
+    HAL_ADC_Start(&hadc1);  // Start the ADC peripheral
+
+    osDelay(5);
```

Comment:
The `Start ADC peripheral` comment is pretty obvious, on the other hand, what this is for is not obvious.